yoti_python_sdk/document_details.py

```diff
@@ -31,14 +26,10 @@ def expiration_date(self):
     def issuing_authority(self):
         return self.__dict__.get("_DocumentDetails__issuing_authority", None)
 
-    def __validate_data(self, data):
-        if self.VALIDATION_REGEX.search(data):
-            return
-        else:
-            raise ValueError("Invalid value for DocumentDetails")
-
     def __parse_data(self, data):
         data = data.split()
+        if len(data) < 3:
+            raise ValueError("Invalid value for DocumentDetails")
```

Comment:
Let's check for empty strings in the list (as this could be indicative of consecutive spaces in the document details string) and add some tests for it (the other SDK's should have examples for this)

Reply:
test_should_parse_with_double_space (test_document_details.py:127) covers this test case. Apparently the implementation of string split on python makes this impossible.

Reply:
The implementation of string split() in Python swallows this. Have added test_should_fail_with_double_space to demonstrate this.